Auth0/JWT+Header.swift

```diff
@@ -0,0 +1,35 @@
+// JWT+Header.swift
+//
+// Copyright (c) 2020 Auth0 (http://auth0.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+import JWTDecode
+
+extension JWT {
```

Comment:
There's no such type as `JWTHeader` in [JWTDecode](https://github.com/auth0/JWTDecode.swift). So I'm adding typed accessors for header properties, following the style fo the `JWTDecode` API.

Reply:
Refer to the lines where you make use of this. You receive a `JWT jwt` and call `jwt.algorithm` or `jwt.kid`, where those are "token header" keys instead, not "token" keys. Thus my comment on renaming this class (if that's possible) or at least the param on the places where you use it.

Reply:
The same could be said about accessing the reserved claims directly from the `JWT` (e.g. `jwt.audience`), since they're part of the "token payload", yet the `JWTDecode` library chooses to expose them like that. Following that API design, I added those two header properties in the same way.